topics/search.md

```diff
@@ -0,0 +1,167 @@
+- No stemming is performed while indexing a tag field.
+- Only prefix, exact pre filters can be performed on a tag field. Suffix, infix queries are not supported.
+- By default, tags are case insensitive. For example, "Blue" and "BLUE" both will be indexed as "blue" and will yield the
+    same result in a hybrid query.
+- Empty strings are neither indexed or queried.
+- During indexing and querying, any trailing whitespace is removed.
+
+**Syntax**
+
+Here `{` and `}` are part of syntax and `|` is used as a OR operator to support multiple tags:
+
+```
+@:{  |  | ...}
+```
+
+For example, the following query will return documents with blue OR black OR green color.
+
+```
+@color:{blue | black | green}
+```
+
+As another example, the following query will return documents containing "hello world" or "hello universe"
+
+```
+@color:{hello world | hello universe}
+```
+
+### Numeric Index
+
+Numeric indexes allow for filtering queries to only return values that are in between a given start and end value.
+
+- Both inclusive and exclusive queries are supported.
+- For open ended queries, `+inf`, `-inf` can be used to express start and end ranges.
+
+As an example, the following query will return books published between 2021 and 2024 (Both inclusive).
+The equivalent mathematical expression is `2021 <= year <= 2024`.
+
+```
+"@year:[2021 2024]"
+```
+
+While The following query will return books published between 2021 (exclusive) and 2024 (inclusive).
+The equivalent mathematical expression is `2021 < year <= 2024`:
+
+```
+"@year:[(2021 2024]"
+```
+
+The following query will return books published before 2024 (inclusive). The equivalent mathematical expression is year `<= 2024`:
+
+```
+@year:[(-inf 2024]
+```
+
+The following query will return books published after 2015 (exclusive). The equivalent mathematical expression is year `>= 2015`:
+
+```
+@year:[2015 +inf]
+```
+
+### Query planner
+
+There are two primary approaches to hybrid queries:
+
+- **Pre-filtering:** Begin by filtering the dataset and then perform an exact similarity search. This works well when the filtered result set is small but can be costly with larger datasets.
+- **Post-filtering:** Perform the similarity search first, then filter the results. This is suitable when the filter-qualified result set is large but may lead to empty or lower than expected amount of results.
+
+Valkey-Search uses a **hybrid approach** with a query planner that selects the most efficient query execution path between:
+
+- **Pre-filtering**
+- **Inline-filtering:** Filters results during the similarity search process.
+
+## Monitoring
+
+To check the server's overall search metrics, you can use the `INFO SEARCH` or `INFO MODULES` commands.
+
+The following metrics are added to the `INFO` command's output:
+
+- **search_used_memory_human** (Integer) A human-friendly readable version of the `search_used_memory_bytes` metric
```

Comment:
```suggestion
- `search_used_memory_human`: A human-friendly readable version of the `search_used_memory_bytes` metric
```
This is more closely related to the format we use for info fields in other places like bloom https://raw.githubusercontent.com/valkey-io/valkey-doc/refs/heads/main/topics/bloomfilters.md or the info command.